app/controllers/application_controller/performance.rb

```diff
@@ -634,7 +634,7 @@ def perf_gen_data_before_wait
                              from_dt,
                              to_dt,
                              interval_type]
-      elsif %w(MiddlewareServer MiddlewareDatasource MiddlewareMessaging).include?(@perf_record.class.name.demodulize)
+      elsif %w(MiddlewareServer MiddlewareServerEap MiddlewareServerWildfly MiddlewareDatasource MiddlewareMessaging).include?(@perf_record.class.name.demodulize)
```

Comment:
We don't want to add every class every time we decide something should be sub-classed. I would prefer this much more: `%w(MiddlewareServer MiddlewareDatasource MiddlewareMessaging).any? { |e| @perf_record.kind_of? e.constantize }`

Reply:
The problem is that this objct is MiddlewareServerEAP so ..or we control that object include MiddlewareServer or we must check the type.

Reply:
Yes I forgot the real-time I'll do after this discussion about how we are going to do it.

Reply:
well it works for me. MiddlewareServerEap is still a `kind_of?` MiddlewareServer so we can check the type properly:

![image](https://user-images.githubusercontent.com/7453394/33727414-38749c4c-db78-11e7-9439-34a65adbb212.png)

The code I've posted works 😉

Reply:
I didn't see that your proposal was kind_of ^^ ok I agree with you, I am gonna make the change Thanks @tumido